lib/events.js

```diff
@@ -42,27 +42,74 @@ exports.onData = function onData(message) {
         // and be reissued again.
       }
     );
-  } else {
-    message.del();
   }
-};
 
-exports.onError = function onError(err) {
-  logger.error('accountEvent', err);
-};
+  function primaryEmailChanged(message) {
+    var userId = message.uid.split('@')[0];
+    if (! HEX_STRING.test(userId)) {
+      message.del();
+      return logger.warn('badPrimaryEmailChanged', {userId: userId});
+    }
 
-exports.start = function start() {
-  if (! config.events.region || ! config.events.queueUrl) {
-    if (env.isProdLike()) {
-      throw new Error('config.events must be included in prod');
+    // We currently a generate a user's cache key from their
+    // credentials in a request object. This builds an object
+    // to match what the server is expecting.
+    var req = {auth: {credentials: {user: userId}}};
+    P.resolve(new Promise(function(resolve) {
```

Comment:
Can anything inside this `new Promise` block fail or throw, e.g. if redis is down or if there's an error talking to SQS?  It may be worth wrapping in a try-catch so that the returned promise will properly error out in such cases.

Also it's not clear why we need to `P.resolve(new Promise())`, we should just be able to use the object created by  `new Promise` directly?

And if it is necessary, we should use either `Promise` or `P` consistently rather than mixing the two :-)